src/sharedentitycommunication.hpp

```diff
@@ -644,18 +652,37 @@ void SharedEntityCommunication<T>::BroadcastSizes(T** mats)
 template <class T>
 void SharedEntityCommunication<T>::BroadcastData(T** mats)
 {
+    const int header_length = size_specifier_ + 1;
     int send_counter = 0;
     int receive_counter = 0;
     data_requests_ = new MPI_Request[num_master_comms_ + num_slave_comms_];
     MPI_Status* data_statuses =
         new MPI_Status[num_master_comms_ + num_slave_comms_];
 
+    // Invert the entity to true entity relation to obtain true entity to entity.
+    // This is the simplest thing to do and costs O(n log n), where n is the number
```

Comment:
With `unordered_map` this cost can be dropped down to `O(n)`, though the difference in performance would probably be negligible.